src/Enzo/utils/utils.cpp

```diff
@@ -0,0 +1,53 @@
+// See LICENSE_CELLO file for license and copyright information
+
+/// @file     utils.cpp
+/// @author   Matthew Abruzzo (matthewabruzzo@gmail.com)
+/// @date     2024-02-20
+/// @brief    Definitions of generic utility functions useful throughout the
+///           Enzo layer
+
+#include "Enzo/enzo.hpp"
+#include "Enzo/utils/utils.hpp"
+
+#include <cmath>
+
+//----------------------------------------------------------------------
+
+static enzo_float median(enzo_float x, enzo_float y, enzo_float z) noexcept
+{
+  // we intentionally use bitwise-and rather than logical-and for speed
+  if ((x >= y) & (x <= z)) {
+    return x;
+  } else if ((x >= z) & (x <= y)) {
+    return x;
+  } else if ((y >= x) & (y <= z)) {
+    return y;
+  } else if ((y >= z) & (y <= x)) {
+    return y;
+  }
+  return z;
+}
+
+bool enzo_utils::consistent_cube_cellwidths(enzo_float dx, enzo_float dy,
+                                            enzo_float dz) noexcept
+{
+  // we want to ensure the median value is within N ULPs of the other 2 values
+  // - a ulp is the difference between 2 adjacent floating point values
```

Comment:
Instead of  "- a ulp is the difference between 2 adjacent floating point values" I suggest you say the following:

A ULP, or 'Unit in the Last Place', is the difference between 2 adjacent floating point values.  Since floating point calculations are not exact and decimal numbers cannot always be represented exactly, comparisons between two nominally identical numbers need to take that into account.  To that end, this routine checks to make sure all values are very close to each other - within N ULPs - and thus it is checking to make sure the three values are approximately equal instead of exactly equal.